### ISSUE TITLE
DataTemplate example doesn't work properly

### ISSUE THREAD

**Issue by Fendoroid** [helix-toolkit/helix-toolkit#407]
There's a major problem with this example: bound properties are set correctly, but non bound properties are not set at all.
To better analyze the problem, i did some edits. I commented out the `FixedElements`'s template:
```
<!--<local:ItemsVisual3D ItemTemplate="{StaticResource Template1}" ItemsSource="{Binding FixedElements}"/>-->
```

And I edited the data template for the `CubeVisual3D`:
```
<local:DataTemplate3D x:Key="Template2">
            <h:CubeVisual3D Center="{Binding Position}" SideLength="2" Fill="OrangeRed"/
</local:DataTemplate3D>
```
With this configuration I still got blue cubes of size 1. Plus, they are strangely misaligned.

However, uncommenting the commented block of code in `DataTemplate3D.cs` apparently sets every property correctly on debugger if you break right before `CreateItem` returns. Strangely, the output is not correct (`Fill` and `SizeLength` are set properly, but not `Center`, that appears to be at the origin).

In all cases, appears the following error:
```
System.Windows.Data Error: 2 : Cannot find governing FrameworkElement or FrameworkContentElement for target element. BindingExpression:Path=Position; DataItem=null; target element is 'CubeVisual3D' (HashCode=61946665); target property is 'Center' (type 'Point3D')
```

I don't know what's happening here, sometimes the results are pretty unpredictable. I think those problems arises because of the binding error. However, I've no idea of how to resolve it.

**Comment by objorke** [comment]
Right, only properties with very simple bindings are currently supported in this example:
https://github.com/helix-toolkit/helix-toolkit/blob/develop/Source/Examples/WPF/ExampleBrowser/Examples/DataTemplate/DataTemplate3D.cs#L46
It should be easy to add support for 'static' values, but still there will be challenges with other types of bindings...

**Comment by Fendoroid** [comment]
The binding error is caused by the binding defined in xaml, the `Content` property of the data template. I still don't know why it shows up, if `CubeModel3d` derives from `Visual3D` which is a `DependencyObject`, that is binding-ready.
If you do a `BindingOperation.ClearBinding` to each bounded property in the `Content` of `DataTemplate3D` the error disappears, but there is no advantage in doing that, since it removes the further binding capabilities of the template.

I still don't get why binding gets invalidated if I set non-bounded properties. That happens only with the foreach loop, not if I set manually single properties like:

```
PropertyInfo sideLengthProperty = typeof(CubeVisual3D).GetProperty("SideLength");
PropertyInfo fillProperty = typeof(CubeVisual3D).GetProperty("Fill");

sideLengthProperty.SetValue(visual, 5f);
fillProperty.SetValue(visual, Brushes.OrangeRed);
```

I'm currently studying advanced data binding documentation hoping to find something useful.
